extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/FSInternetRadioHandlerOSGiTest.java

```diff
@@ -0,0 +1,969 @@
+        ChannelUID channelUID = new ChannelUID(thingUID, channelID);
+        ChannelTypeUID channelTypeUID = new ChannelTypeUID(FSInternetRadioBindingConstants.BINDING_ID,
+                channelUID.getIdWithoutGroup());
+        Channel radioChannel = new Channel(channelUID, acceptedItemType);
+        channels.add(radioChannel);
+        return radioChannel;
+    }
+
+    private void testRadioThingConsideringConfiguration(Thing thing) {
+        Configuration config = thing.getConfiguration();
+        if (isConfigurationComplete(config)) {
+            waitForAssert(() -> {
+                assertThat("The radioThing with complete configuration was not created successfully", thing.getStatus(),
+                        is(equalTo(ThingStatus.ONLINE)));
+            });
+        } else {
+            waitForAssert(() -> {
+                assertThat(
+                        "The ThingStatus was not updated correctly when a Thing with incomplete configuration was created",
+                        thing.getStatus(), is(equalTo(ThingStatus.OFFLINE)));
+                assertThat(
+                        "The ThingStatusInfo was not updated correctly when a Thing with incomplete configuration was created",
+                        thing.getStatusInfo().getStatusDetail(), is(equalTo(ThingStatusDetail.CONFIGURATION_ERROR)));
+            });
+        }
+    }
+
+    private boolean isConfigurationComplete(Configuration config) {
+        String ip = (String) config.get(FSInternetRadioBindingConstants.CONFIG_PROPERTY_IP);
+        BigDecimal port = (BigDecimal) config.get(FSInternetRadioBindingConstants.CONFIG_PROPERTY_PORT.toString());
+        String pin = (String) config.get(FSInternetRadioBindingConstants.CONFIG_PROPERTY_PIN.toString());
+
+        if (ip == null || port.compareTo(BigDecimal.ZERO) == 0 || StringUtils.isEmpty(pin)) {
+            return false;
+        }
+        return true;
+    }
+
+    private Thing initializeRadioThing(Configuration config) {
+        Thing radioThing = ThingBuilder.create(DEFAULT_THING_TYPE_UID, DEFAULT_THING_UID).withConfiguration(config)
+                .withChannels(channels).build();
+        managedThingProvider.add(radioThing);
+
+        radioHandler = waitForAssert(() -> {
+            final ThingHandler thingHandler = radioThing.getHandler();
+            assertThat("RadioHandler service is not found", thingHandler, is(instanceOf(FSInternetRadioHandler.class)));
+            return (FSInternetRadioHandler) thingHandler;
+        });
+        return radioThing;
+    }
+
+    private void turnTheRadioOn(Thing radioThing) {
+        radioHandler.handleCommand(radioThing.getChannel(FSInternetRadioBindingConstants.CHANNEL_POWER).getUID(),
+                OnOffType.ON);
+
+        FrontierSiliconRadio radio;
+        try {
+            Field f = radioHandler.getClass().getDeclaredField("radio");
+            f.setAccessible(true);
+            radio = (FrontierSiliconRadio) f.get(radioHandler);
```

Comment:
I tried to remove all reflection used by the Groovy tests.
The only solution for this one has been to change the access level for the radio member to "no modifier" or protected.
If give me a go to change the access level we could drop that reflection.